stdlib/Random/src/MersenneTwister.jl

```diff
@@ -568,17 +576,16 @@ function _randjump(r::MersenneTwister, jumppoly::DSFMT.GF2X)
     s
 end
 
-# NON-PUBLIC
-function jump(r::MersenneTwister, steps::Integer)
-    iseven(steps) || throw(DomainError(steps, "steps must be even"))
-    # steps >= 0 checked in calc_jump (`steps >> 1 < 0` if `steps < 0`)
-    j = _randjump(r, Random.DSFMT.calc_jump(steps >> 1))
-    j.adv_jump += steps
-    j
+function jump(rng::MersenneTwister; by::Real=NaN)
+    isnan(by) && (by = 2.0^128)
```

Comment:
Why not set the default value for `by` to be 2.0^128 rather than `NaN` if we're already changing the value to be that when the default is provided?

Reply:
Good point.  The weak reason is that the array version takes the same default `NaN`, and passes it out unchanged to the non-array version (this method above), which then needs to handle `NaN` anyway. The alternative would be, in the array version, to either call `jump(rng)` or `jump(rng; by)` depending on whether or not `by` was passed explicitly.